### ISSUE TITLE
Improve v2  Error Message "ValueError: max() arg is an empty sequence"

### ISSUE THREAD

**Issue by zaro0508** [cloudreach/sceptre#575]
We've been using sceptre v 1.x and it's been working great.  I tried testing v 2.0.1 and I'm getting "ValueError: max() arg is an empty sequence" when running any command.  I've tried running sceptre on python 2.7 and 3.6 getting that error in both versions.  I also tried running from master and from 2.0.1 release tag and still getting the same error.

My sceptre test project: [scpester-infra.zip](https://github.com/cloudreach/sceptre/files/2721498/scpester-infra.zip)

Install and run log: [sceptre2_run.log](https://github.com/cloudreach/sceptre/files/2721509/sceptre2_run.log)

 
(py36-sceptredev) ~/w/scpester-infra ❯❯❯ sceptre --var "profile=sandbox.cfservice" validate config/test/noop.yaml        
Traceback (most recent call last):
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/bin/sceptre", line 11, in <module>
    load_entry_point('sceptre', 'console_scripts', 'sceptre')()
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/zaro0508/.virtualenvs/py36-sceptredev/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/zaro0508/work-sage/sceptre/sceptre/cli/helpers.py", line 37, in decorated
    return func(*args, **kwargs)
  File "/Users/zaro0508/work-sage/sceptre/sceptre/cli/template.py", line 35, in validate_command
    responses = plan.validate()
  File "/Users/zaro0508/work-sage/sceptre/sceptre/plan/plan.py", line 318, in validate
    return self._execute(*args)
  File "/Users/zaro0508/work-sage/sceptre/sceptre/plan/plan.py", line 36, in _execute
    executor = SceptrePlanExecutor(self.command, self.launch_order)
  File "/Users/zaro0508/work-sage/sceptre/sceptre/plan/executor.py", line 35, in __init__
    self.num_threads = len(max(launch_order, key=len))
ValueError: max() arg is an empty sequence

**Comment by ngfgrant** [comment]
@zaro0508 thanks for providing the info and the test project - I will have a look and get back to you. That error however raises when stacks have not been able to be read. My first guess is that there are some incorrect values in your test project config and v2 is then not able to recognise them as valid stacks; therefore they are not able to be loaded into the graph launch order. 

I should be able to get some more information when I get a moment to look at your project.

**Comment by allumbra** [comment]
I'm getting the same error

**Comment by rafcio19** [comment]
This issue will appear if the stack config file does not exist, ie. if the `config` folder looks like:

```
config
├── config.yaml
└── dev
    ├── config.yaml
    ├── stack1.yaml
    ├── stack2.yaml
    ├── stack3.yaml

```

then the command should be `sceptre validate dev/stack1.yaml`. 

@zaro0508 , in your case it looks like you need to run `sceptre --var "profile=sandbox.cfservice" validate test/noop.yaml`.

@ngfgrant please improve `ValueError: max() arg is an empty sequence` error message, it is not very helpful.

**Comment by ngfgrant** [comment]
@rafcio19 Thanks for adding to the conversation. Agreed that the message is not good, working on improving it in an upcoming patch release 👍

**Comment by priorax** [comment]
I've tried doing similar with the following folder structure
```
config
 ├── config.yaml
 └── dev
     ├── config.yaml
     ├── stack1.yaml
     ├── stack2.yaml
```
Based on the feedback of @rafcio19 
If I run: `sceptre validate dev`
Then stack1 and stack2 validate fine
If I do `sceptre validate dev/stack1.yaml` or `sceptre validate dev/stack1`
Then I still get the max() arg is an empty sequence error.

While I'm not doubting that the error needs to get fixed, is there any recommendations on why that is occurring?

**Comment by mMeijden** [comment]
Arriving bit late to the party maybe, but i'm new to sceptre so i was following the tutorial (https://sceptre.cloudreach.com/latest/docs/get_started.html) 
where the same error comes up as well.
`$ sceptre create dev/vpc.yaml` is where it get's stuck.
`$ sceptre create vpc.yaml` works fine
using python 3.7

**Comment by ngfgrant** [comment]
So we don't officially "support" python > 3.6. @mMeijden what is the `tree` of your Sceptre project.

**Comment by mike-rsi** [comment]
@mMeijden I found there was problems with paths in Windows in the current release. Does this comment help you? https://github.com/cloudreach/sceptre/issues/531#issuecomment-453259989

**Comment by mike-rsi** [comment]
@mMeijden Ah, yes, I am explicitly naming my stacks in my configs.  I haven't tried it while letting Sceptre to auto generate the names, but I could imagine the same problems appears there as well.

**Comment by ngfgrant** [comment]
Hey mike-rsi - Use back slashes for windows (for now, there is a ticket to improve) and read the docs on using custom stack names to see if that would impact your setup https://sceptre.cloudreach.com/latest/docs/stack_config.html#stack_name

**Comment by jamielennox** [comment]
I hit this issue because I did `sceptre launch config/dev` instead of `sceptre launch dev` and so it didn't find the right template. This should be a pretty easy stack not found message.

**Comment by ngfgrant** [comment]
Yep 
https://github.com/cloudreach/sceptre/commit/244288ea25274db4a542eacff489fb6f26cc98aa resolves this. Will be released in 2.0.3.

**Comment by et304383** [comment]
Getting this error when using a symlink for the config path.  

```
tree
.
├── config
│   ├── base
│   │   ├── config.yaml
│   │   └── dynamo.yaml
│   ├── config.yaml
│   └── dev -> base
└── templates
    └── dynamo.j2
```

So sceptre 2 broke symlinks on config dirs too?  My goodness, V2 is less than ideal.
